README.Rmd

```diff
@@ -8,92 +8,159 @@ params:
+An example of a shiny app that uses `shiny.info` can be found in `./examples` directory.
+
+![](inst/assets/README_files/example.gif)
+
+Basic features
+-------
+
+* display a simple text message:
+  ```
+  shiny.info::display("Hello user!", position = "top right")
+  ```
+  ![](inst/assets/README_files/display.png)
+* show information about git branch, commit and changes:
+  ```
+  shiny.info::git_info()
+  ```
+  ![](inst/assets/README_files/git.png)
+* add "powered by" information with link:
+  ```
+  shiny.info::powered_by("Appsilon", link = "appsilon.com")
+  ```
+  ![](inst/assets/README_files/powered.png)
+* show app version:
+  ```
+  # global variable:
+  VERSION <- "1.2.2"
+
+  # in app ui
+  shiny.info::version()
+  ```
+  ![](inst/assets/README_files/version.png)
+* show a busy spinner when app is calculating:
+  ```
+  shiny.info::busy()
+  ```
+  ![](inst/assets/README_files/busy.gif)
+* group multiple messages in one panel:
+  ```
+  shiny.info::info_panel(
+      shiny.info::git_info(),
+      shiny.info::powered_by("Appsilon", link = "appsilon.com"),
+      position = "bottom left"
     )
-  ),
-  server = function(input, output, session) {
-
-    # Combine the selected variables into a new data frame
-    selectedData <- reactive({
-      iris[, c(input$xcol, input$ycol)]
-    })
-
-    clusters <- reactive({
-      kmeans(selectedData(), input$clusters)
-    })
-
-    output$plot1 <- renderPlot({
-      palette(c("#E41A1C", "#377EB8", "#4DAF4A", "#984EA3",
-                "#FF7F00", "#FFFF33", "#A65628", "#F781BF", "#999999"))
-
-      par(mar = c(5.1, 4.1, 0, 1))
-      plot(selectedData(),
-           col = clusters()$cluster,
-           pch = 20, cex = 3)
-      points(clusters()$centers, pch = 4, cex = 4, lwd = 4)
-    })
-
-  }
-)
+  ```
+  ![](inst/assets/README_files/panel.png)
+
+Advanced features
+-------
+
+* render value (eg. input, reactive value) from the server:
+  ```
+  # in app ui
+  shiny.info::info_value("test_info_value")
+
+  # in app server
+  some_value <- reactiveVal("a test value to display")
+  output$test_info_value <- shiny.info::render_info_value(some_value())
+  ```
+  ![](inst/assets/README_files/info_value.png)
+* render informations about the session:
+  ```
+  # in app ui
+  shiny.info::info_value("session_info_value")
+
+  # in app server
+  output$session_info_value <- shiny.info::render_session_info()
+  ```
+  ![](inst/assets/README_files/session.png)
+* debug app using `browser()` function just by clicking a button:
+  ```
+  # in app ui
+  shiny.info::inspect_btn_ui()
+
+  # in app server
+  shiny.info::inspect_btn_server(input)
+  ```
+  ![](inst/assets/README_files/inspect_button.png)
+* toggle display with a key shortcut:
+  ```
+  shiny.info::toggle_info("Ctrl+Shift+K")
+  ```
+  ![](inst/assets/README_files/shortcut.gif)
+
+How to contribute?
+-----------------
+If you want to contribute to this project please submit a regular PR, once you're done with new feature or bug fix.<br>
```

Comment:
If you want to contribute to this project please submit a regular PR once you're done with your new feature or bug fix.<br>